x-pack/plugins/infra/public/pages/logs/log_entry_rate/use_log_entry_anomalies_results.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useMemo, useState, useCallback, useEffect } from 'react';
+
+import { LogEntryAnomaly } from '../../../../common/http_api';
+import { useTrackedPromise } from '../../../utils/use_tracked_promise';
+import { callGetLogEntryAnomaliesAPI } from './service_calls/get_log_entry_anomalies';
+import { Sort, Pagination, PaginationCursor } from '../../../../common/http_api/log_analysis';
+
+export type SortOptions = Sort;
+export type PaginationOptions = Pick<Pagination, 'pageSize'>;
+export type Page = number;
+export type FetchNextPage = () => void;
+export type FetchPreviousPage = () => void;
+export type ChangeSortOptions = (sortOptions: Sort) => void;
+export type ChangePaginationOptions = (paginationOptions: PaginationOptions) => void;
+export type LogEntryAnomalies = LogEntryAnomaly[];
+interface PaginationCursors {
+  previousPageCursor: PaginationCursor;
+  nextPageCursor: PaginationCursor;
+}
+
+export const useLogEntryAnomaliesResults = ({
+  endTime,
+  startTime,
+  sourceId,
+  lastChangedTime,
+  defaultSortOptions,
+  defaultPaginationOptions,
+}: {
+  endTime: number;
+  startTime: number;
+  sourceId: string;
+  lastChangedTime: number;
+  defaultSortOptions: Sort;
+  defaultPaginationOptions: Pick<Pagination, 'pageSize'>;
+}) => {
+  // Pagination
+  const [page, setPage] = useState(1);
```

Comment:
Yes, true.

Ultimately if we want to show the page number, like it is now, someone needs to track it, but it could definitely be lifted out from the hook and data concerns, and made exclusively a UI concern.

Alternatively, there could just be next and previous buttons, and we remove showing the page number.

Reply:
The page number doesn't add much IMHO, so just the next and previous buttons should be fine. Maybe a "|< first" button could be useful too?

But it's cosmetic and not something that should keep this PR from being merged, I think.